bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolExecutorTest.java

```diff
@@ -0,0 +1,45 @@
+package org.eclipse.smarthome.core.scheduler;
+
+import static org.junit.Assert.*;
+
+import java.text.ParseException;
+
+import org.eclipse.smarthome.core.scheduler.ExpressionThreadPoolManager.ExpressionThreadPoolExecutor;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class ExpressionThreadPoolExecutorTest {
+
+    private ExpressionThreadPoolExecutor scheduler = ExpressionThreadPoolManager.getExpressionScheduledPool("test");
+    protected boolean success;
+
+    @Test
+    @Ignore
+    public void testScheduleAndRemove() throws ParseException, InterruptedException {
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                success = true;
+            }
+        };
+
+        success = false;
+        scheduler.schedule(runnable, new CronExpression("0/1 * * * * ?"));
+        Thread.sleep(1500);
+        assertTrue(success);
+
+        boolean removed = scheduler.remove(runnable);
+
+        // TODO: This currently fails - why?
```

Comment:
The test succeeds if I change the return line to
```
            return super.remove((Runnable) future);
```
So here again, the super implementation uses the future in the workQueue.